x-pack/platform/plugins/private/product_intercept_trigger/server/core.ts

```diff
@@ -213,9 +223,24 @@ export class ProductInterceptTriggerCore {
     this.taskManager = taskManager;
 
     const existingTask = await this.fetchRegisteredTriggerTask();
+    let shouldReschedule = false;
+
+    if (
+      this.isCloudDeployment &&
+      existingTask &&
+      existingTask.state.installedOn !== this.ctx.env.packageInfo.version
```

Comment:
Note to self comparing string doesn't suffice here, actually verify with a semver util